### ISSUE TITLE
Prompt users for unknown variables/placeholders in configurations before starting a debugging session

### ISSUE THREAD

**Issue by Frederick888** [mfussenegger/nvim-dap#501]
### Problem Statement

I'm using nvim-dap with Go (Delve DAP).

Some of our test files have build tags [1][2] in them, so they are not compiled in test binaries by default. Delve handles this by allowing clients to provide build flags on connection. For example, currently I have these configurations for no tags, `unit` tag, `component` tag respectively:

```lua
dap.configurations.go = {
  {
    type = "go",
    name = "local-test",
    request = "launch",
    mode = "test",
    program = "./${relativeFileDirname}",
    args = { "-test.v", },
  },
  {
    type = "go",
    name = "local-test-unit",
    request = "launch",
    mode = "test",
    program = "./${relativeFileDirname}",
    args = { "-test.v", },
    buildFlags = "-gcflags='all=-N -l' -tags=unit",
  },
  {
    type = "go",
    name = "local-test-component",
    request = "launch",
    mode = "test",
    program = "./${relativeFileDirname}",
    args = { "-test.v", },
    buildFlags = "-gcflags='all=-N -l' -tags=component",
  },
}
```

But this list can grow quickly when I have more tags.

Ideally I'd like to have a single configuration like:

```lua
dap.configurations.go = {
  {
    type = "go",
    name = "local-test-with-tags",
    request = "launch",
    mode = "test",
    program = "./${relativeFileDirname}",
    args = { "-test.v", },
    buildFlags = "-gcflags='all=-N -l' -tags=${buildTags}",
  },
}
```

...and it asks me for `${buildTags}`'s value before starting the debugging session.

[1] https://pkg.go.dev/cmd/go@go1.16.15#hdr-Build_constraints
[2] https://pkg.go.dev/cmd/go@go1.18#hdr-Build_constraints

### Ideas or possible solutions

This particular issue can be resolved by customising my `dap.adapters.go` function I guess, but imho it's nice to have it built-in to tackle issues like:

- condition builds
- providing different environment variable values for each run
- other kinds of configurations that a DAP server allows

**Comment by Frederick888** [comment]
@mfussenegger That worked like a charm! I didn't know you could do that. May I know if this is some Lua feature or an nvim-dap one?

**Comment by mfussenegger** [comment]
> May I know if this is some Lua feature or an nvim-dap one?

It's a nvim-dap feature. See https://github.com/mfussenegger/nvim-dap/blob/c20c78d7c6c82f16a2d1abec31f4273194e3987b/lua/dap.lua#L139-L141
